src/blue.ts

```diff
@@ -259,13 +263,14 @@ export function blueProxyFactory(env: MembraneBroker) {
             if (isUndefined(redDescriptor)) {
                 // looking in the blue proto chain to avoid switching sides
                 const blueProto = getBlueValue(ReflectGetPrototypeOf(target));
-                return ReflectSet(blueProto, key, value, receiver);
-            }
-            if (hasOwnProperty(redDescriptor, 'set')) {
+                if (!isNull(blueProto)) {
```

Comment:
if the blueProto is null, we go with the last line in the method to those the reflect.set instead.

Reply:
a potential problem here is a shape shifter that adds the proto after saying that it doesn't have a proto. but still pretty safe.

Reply:
> a potential problem here is a shape shifter that adds the proto after saying that it doesn't have a proto. but still pretty safe.

I would hope that would trigger some kind of invariant error.